packages/contracts/contracts/minter-suite/Minters/one-off/ClaimMinter.sol

```diff
@@ -0,0 +1,452 @@
+     * @param timestampStart_ Unix timestamp when claiming can begin.
+     */
+    function configureTimestampStart(uint256 timestampStart_) external {
+        // CHECKS
+        AuthLib.onlyCoreAdminACL({
+            coreContract: coreContractAddress,
+            sender: msg.sender,
+            contract_: address(this),
+            selector: this.configureTimestampStart.selector
+        });
+
+        // EFFECTS
+        timestampStart = timestampStart_;
+    }
+
+    /**
+     * @notice Pre-mints `amount` tokens to this contract
+     * for later claiming by collectors. Only callable by the core contract's Admin ACL.
+     * @dev This function mints tokens directly to this contract address, marking them
+     * as pre-minted but not yet claimed. The tokens remain in this contract's possession
+     * until claimed by users through the claimToken function. This allows for controlled
+     * distribution of tokens with arbitrary claiming order.
+     * @param amount Number of tokens to pre-mint.
+     */
+    function preMint(uint256 amount) external nonReentrant {
+        // CHECKS
+        AuthLib.onlyCoreAdminACL({
+            coreContract: coreContractAddress,
+            sender: msg.sender,
+            contract_: address(this),
+            selector: this.preMint.selector
+        });
+
+        MaxInvocationsLib.preMintChecks({
+            projectId: projectId,
+            coreContract: coreContractAddress
+        });
+
+        // EFFECTS
+        // Mint tokens to this contract
+        for (uint256 i = 0; i < amount; i++) {
+            uint256 tokenId = minterFilter.mint_joo({
+                to: address(this),
+                projectId: projectId,
+                coreContract: coreContractAddress,
+                sender: msg.sender
+            });
+
+            MaxInvocationsLib.validateMintEffectsInvocations({
+                tokenId: tokenId,
+                coreContract: coreContractAddress
+            });
+            // Mark token as pre-minted
+            _markTokenPreMinted(tokenId);
+        }
+    }
+
+    /**
+     * @notice Claims token `tokenId` by paying the required price.
+     * Available once claiming has started.
+     * @dev This function allows users to claim pre-minted tokens by paying the exact price
+     * calculated by priceByTokenIdInWei. The function checks that the token is not already
+     * claimed, that claiming has started, and that the exact payment
+     * amount is provided. Upon successful claim, the token is marked as claimed, configured
+     * with a pseudorandom hash seed via PMP, revenue is split automatically, and the token
+     * is transferred to the claimant. This enables arbitrary claiming order where users can
+     * claim any available token ID in any sequence.
+     * @param tokenId Token ID to claim.
+     */
+    function claimToken(uint256 tokenId) external payable nonReentrant {
+        // check that token id is unclaimed
```

Comment:
nit: add comment explaining how valid token Ids will be operationally handled by frontend and admin pre-minting

Reply:
added in https://github.com/ArtBlocks/artblocks-contracts/pull/1857/commits/0a0aab231242b8eca8df6c50d3a99b255efd9e45